### PR TITLE
Fix test suite when running on MPS-enabled hardware

### LINKED ISSUE #14012: Update tests that mock cuda availability and extend it for mps
## Proposed refactor

We have tests that fail locally on MPS supported devices because their mocking of CUDA does not extend to MPS. After #13947 landed, the availability check actually works now and we need to update some tests. These are mainly the tests that have this combination of settings:

- has `accelerator="gpu"` (which includes both cuda and mps)
- has function mocks like `torch.cuda.is_available` as True or `torch.cuda.device_count > 0`. 

An example of such a test:

https://github.com/Lightning-AI/lightning/blob/e6a8283e9cd9df53fb661c64bbf2037e1391a16d/tests/tests_pytorch/trainer/connectors/test_accelerator_connector.py#L245-L262

### Motivation

Let MPS users run the tests locally (me).

### Pitch

Extend the mocking to MPS where applicable. In some tests it may make sense to actually only test for cuda, in which case we should change `accelerator="gpu"` to `"cuda"`. Some other tests may need to be skipped entirely on MPS. 

Note that simply slapping a `RunIf(min_cuda=x)` on top of these tests is not an option, as for most of the tests we want to run them with mocks on the CPU. 

### Additional context

Rather do this sooner than later. Right now, I can't differentiate from tests failing because of bugs in my branch vs. them failing due to this. 

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

cc @borda @akihironitta @justusschock

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #14012

Simplifies my life. With all tests running properly on M1 Macs, the developer can be sure that if a test fails, it is not just because they are on M1. 

I understand that we don't have CI runners for M1, so some tests may get outdated over time. But this is still a valuable quality of life improvement. 

### Does your PR introduce any breaking changes? If yes, please list them.

No.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @akihironitta @justusschock

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by otaj — APPROVED** [review]

**Review by Borda — APPROVED** [review]
@awaelchli, have you tested it locally that all passes? :)

**Comment by awaelchli** [comment]
@Borda Yes, ran all tests under tests/tests_pytorch and tests/tests_lite. 
Will do again when doing merge conflict resolution.
